v0/destinations/user/transform.js

```diff
@@ -0,0 +1,212 @@
+  createEventOccurrencePayloadBuilder
+} = require("./utils");
+
+const { EventType } = require("../../../constants");
+
+const responseBuilder = async (payload, endpoint, method, apiKey) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = endpoint;
+    response.headers = {
+      "Content-Type": "application/json",
+      Authorization: `Token ${apiKey}`,
+      Accept: "*/*;version=2"
+    };
+    response.method = method;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new CustomError("Payload could not be constructed", 400);
+};
+
+const identifyResponseBuilder = async (message, destination) => {
+  let payload;
+  let endpoint;
+  let method;
+  let builder;
+  const user = await retrieveUserFromLookup(message, destination);
+  const { Config } = destination;
+  const { apiKey, appSubdomain } = Config;
+  // If user already exist we will update it else creates a new user
+  if (!user) {
+    builder = createUserPayloadBuilder(message, destination);
+    payload = builder.payload;
+    endpoint = builder.commonUserPropertiesPayload;
+    method = builder.method;
+  } else {
+    const { id } = user;
+    builder = updateUserPayloadBuilder(message, destination);
+    payload = builder.commonUserPropertiesPayload;
+    endpoint = builder.endpoint.replace("<user_id>", id);
+    method = builder.method;
+  }
+  endpoint = prepareUrl(endpoint, appSubdomain);
+  return responseBuilder(payload, endpoint, method, apiKey);
+};
+
+const trackResponseBuilder = async (message, destination) => {
+  if (!message.event) {
+    throw new CustomError("parameter event is required", 400);
+  }
+
+  let payload;
+  let endpoint;
+  let method;
+  let builder;
+  const user = await retrieveUserFromLookup(message, destination);
+  const { Config } = destination;
+  const { apiKey, appSubdomain } = Config;
+  if (user) {
+    builder = createEventOccurrencePayloadBuilder(message, user, destination);
+    payload = builder.payload;
+    endpoint = builder.endpoint;
+    method = builder.method;
+    endpoint = prepareUrl(endpoint, appSubdomain);
+    return responseBuilder(payload, endpoint, method, apiKey);
+  }
+
+  throw new CustomError("No user found with given lookup field", 400);
+};
+
+const pageResponseBuilder = async (message, destination) => {
+  let payload;
+  let endpoint;
+  let method;
+  let builder;
+  const user = await retrieveUserFromLookup(message, destination);
+  const { Config } = destination;
+  const { apiKey, appSubdomain } = Config;
+  if (user) {
+    builder = pageVisitPayloadBuilder(message, user);
+    payload = builder.payload;
+    endpoint = builder.endpoint;
+    method = builder.method;
+    endpoint = prepareUrl(endpoint, appSubdomain);
+    return responseBuilder(payload, endpoint, method, apiKey);
+  }
+  throw new CustomError("No user found with given lookup field", 400);
```

Comment:
Lets write "No user found with given lookup field. Page event cannot be completed if there is no valid user". Also let us re-format the error messages, let's take ref from clickup and others.